packages/app/src/systems/DApp/pages/ConnectionRequest/ConnectionRequest.tsx

```diff
@@ -71,23 +63,15 @@ export function ConnectionRequest() {
                   </Card.Header>
                   <Card.Body css={styles.accountCardBody}>
                     {accounts?.map((account) => {
-                      const { address, name } = account;
+                      const { address } = account;
                       const isConnected = handlers.isAccountSelected(address);
-                      const rightEl = (
-                        <Flex css={styles.switchWrapper}>
-                          <Switch
-                            size="sm"
-                            checked={isConnected}
-                            aria-label={`Toggle ${name}`}
-                            onCheckedChange={() =>
-                              handlers.toggleAccount(address)
-                            }
-                          />
-                        </Flex>
-                      );
                       return (
                         <motion.div key={address} {...animations.slideInTop()}>
-                          <AccountItem account={account!} rightEl={rightEl} />
+                          <AccountItem
+                            account={account!}
+                            onToggle={() => handlers.toggleAccount(address)}
```

Comment:
can we make the `AccountItem` component pass its own `account.address` ? same for `onUpdate`